### PR TITLE
[v7] Group capabilities

### PR DESCRIPTION
**Pull request by doctrino**
## Description
Please describe the change you have made.

## Checklist:
- [x] Tests added/updated.
- [x] Documentation updated. Documentation is generated from docstrings - these must be updated according to your change.
  If a new method has been added it should be referenced in [cognite.rst](https://github.com/cognitedata/cognite-sdk-python/blob/master/docs/source/cognite.rst) in order to generate docs based on its docstring.
- [x] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-sdk-python/blob/master/CHANGELOG.md).
- [ ] Version bumped. If triggering a new release is desired, bump the version number in [_version.py](https://github.com/cognitedata/cognite-sdk-python/blob/master/cognite/client/_version.py) and [pyproject.toml](https://github.com/cognitedata/cognite-sdk-python/blob/master/pyproject.toml) per [semantic versioning](https://semver.org/).

### CODE REVIEW

File: cognite/client/data_classes/capabilities.py [hunk 1]
```diff
@@ -0,0 +1,630 @@
...
+        }
+        if isinstance(self, UnknownAcl):
+            capability_name = self.capability_name
+        else:
+            capability_name = self._capability_name
+        return {to_camel_case(capability_name) if camel_case else to_snake_case(capability_name): data}
+
+
+@dataclass
+class UnknownAcl(Capability):
```
Comment (erlendvollset) [inline cognite/client/data_classes/capabilities.py:169]: Should this not have actions aswell? Could be just a list of strings.

File: cognite/client/data_classes/capabilities.py [hunk 2]
```diff
@@ -0,0 +1,748 @@
...
+
+    class Action(enum.Enum):
+        Read = "READ"
+        Write = "WRITE"
+
+    class Scope:
+        All = AllScope
+
+    actions: Sequence[Action]
+    scope: AllScope = field(default_factory=AllScope)
```
Comment (erlendvollset) [inline cognite/client/data_classes/capabilities.py:422]: This is not needed. This service was removed a long time ago. I'll update the docs to reflect it.

File: tests/tests_integration/test_api/test_iam.py [hunk 3]
```diff
@@ -1,26 +1,42 @@
...
         metadata = {"haha": "blabla"}
-        group = cognite_client.iam.groups.create(
-            Group(
-                name="bla", capabilities=[{"eventsAcl": {"actions": ["READ"], "scope": {"all": {}}}}], metadata=metadata
+        group: Group | None = None
+        try:
+            group = cognite_client.iam.groups.create(
+                Group(
+                    name="bla",
+                    capabilities=[EventsAcl([EventsAcl.Action.Read], EventsAcl.Scope.All())],
```
Comment (erlendvollset) [inline tests/tests_integration/test_api/test_iam.py:31]: Very nice!

**Review by erlendvollset — APPROVED** [review]

File: cognite/client/utils/_text.py [hunk 4]
```diff
@@ -21,7 +21,8 @@ def random_string(size: int = 100, sample_from: str = string.ascii_uppercase + s
 @lru_cache(maxsize=128)
 def to_camel_case(snake_case_string: str) -> str:
     components = snake_case_string.split("_")
-    return components[0] + "".join(x.title() for x in components[1:])
+    output = components[0] + "".join(x.title() for x in components[1:])
+    return output[0].casefold() + output[1:]
```
Comment (doctrino) [inline cognite/client/utils/_text.py:25]:
```suggestion
    return components[0] + "".join(x.title() for x in components[1:])
```

**Pull request merged**